src/main/scala/beam/agentsim/agents/ridehail/RideHailManager.scala

```diff
@@ -1712,7 +1707,7 @@ class RideHailManager(
       .identifyAdditionalVehiclesForRefuelingDuringContinueRepositioningAndAssignDepotHook(idleVehicles, tick)
 
     val vehiclesWithoutCustomVehicles = idleVehicles.filterNot {
-      case (vehicleId, _) => additionalCustomVehiclesForDepotCharging.contains(vehicleId)
+      case (vehicleId, _) => additionalCustomVehiclesForDepotCharging.map(_._1).contains(vehicleId)
```

Comment:
This is an actual logic change as it was flagged as a mismatch on type checks. It would always be false